lib/dev/ui/spinner.rb

```diff
@@ -62,8 +64,18 @@ def check
           end
 
           def render(index, force = true)
-            return full_render(index) if force
+            should_full_render = force || @force_full_render
+            return full_render(index) if should_full_render
             partial_render(index)
+          ensure
+            @force_full_render = false
+          end
+
+          def update_title(new_title, pad_to_cover: true)
+            @title = new_title
+            @title += right_pad if pad_to_cover
```

Comment:
you can probably just emit `\e[K` instead to clear from the cursor to the end of the line: https://en.wikipedia.org/wiki/ANSI_escape_code